flink-connectors/flink-connector-elasticsearch7/src/main/java/org/apache/flink/streaming/connectors/elasticsearch7/Elasticsearch7UpsertTableSink.java

```diff
@@ -185,8 +191,18 @@ protected ElasticsearchUpsertTableSinkBase copy(
 		Optional.ofNullable(sinkOptions.get(BULK_FLUSH_BACKOFF_DELAY))
 			.ifPresent(v -> builder.setBulkFlushBackoffDelay(Long.valueOf(v)));
 
-		builder.setRestClientFactory(
-			new DefaultRestClientFactory(sinkOptions.get(REST_PATH_PREFIX)));
+		if (Optional.ofNullable(sinkOptions.get(CONNECTOR_USERNAME)).isPresent() &&
+			Optional.ofNullable(sinkOptions.get(CONNECTOR_PASSWORD)).isPresent()) {
+			builder.setRestClientFactory(new AuthRestClientFactory(
+				sinkOptions.get(CONNECTOR_USERNAME),
+				sinkOptions.get(CONNECTOR_PASSWORD),
+				sinkOptions.get(REST_PATH_PREFIX)
+			));
+		} else {
+			builder.setRestClientFactory(
+				new DefaultRestClientFactory(
```

Comment:
I just wondering why es7 doesn't support `REST_MAX_RETRY_TIMEOUT`, I guess it is a mistake of FLINK-13025. Could you add the retry support for es7? cc @twalthr @yanghua

Reply:
@wuchong es7 RestClientBuilder don't have maxRetryTimeout fields